### ISSUE TITLE
d_module_versions & co. only get applied to the first file

### ISSUE THREAD

**Issue by ximion** [mesonbuild/meson#3337]
Hi!
A Meson snippet like this in Meson 0.45:
```meson
test_src = [
    'source/aaa.d',
    'source/bbb.d'
]

src_dir = include_directories('source/')

test_lib = library('test-lib',
        [test_src],
        include_directories: [src_dir],
        install: true,
        version: meson.project_version(),
        soversion: project_soversion,
        d_module_versions: ['Use_Static']
)
```
produces the following (wrong Ninja file:
```ninja
build test@sha/source_derelict_pq_pq.d.o: d_COMPILER ../source/aaa.d
 DEPFILE = test@sha/source_aaa.d.o.deps
 ARGS = -Itest@sha -I. -I.. -I../source/ -enable-color -wi -g -O0 -relocation-model=pic -d-version=Use_Static

build test@sha/source_bbb.d.o: d_COMPILER ../source/bbb.d
 DEPFILE = test@sha/source_bbb.d.o.deps
 ARGS = -Itest@sha -I. -I.. -I../source/ -enable-color -wi -g -O0 -relocation-model=pic
```
Notice the `-d-version=Use_Static` is missing from the second build target.
Because thze unittest in Meson for this has only one file, this issue so far was no uncovered, but is hitting projects in the wild now.
Using attributes like `d_args` works as expected and flags added there are applied to all files that make up a target.

I looked around in the code for a bit, but couldn't find anything unusual. This for sure broke with Meson 0.45 though.

**Comment by ximion** [comment]
This is really bad, because it also causes unit tests to not run. Git bisect says that this was broken in commit ea3b54d40252fcb87eb1852223f125398b1edbdf
I added a new PR with some test cases for this issue.@jpakkane, if you have the time, could you maybe take a look at this?
Currently, it is unclear to me why the flags are only applied to one target, and the commit that introduced this change is quite big (and this might be a really quick and obvious change for people who know the code better).

**Comment by ximion** [comment]
Fixed in 2dc1e87caeeefa5f15f0b8b6454ee65c18012c06
